pilot/pkg/features/pilot.go

```diff
@@ -383,7 +383,7 @@ var (
 	ClusterName = env.RegisterStringVar("CLUSTER_ID", "Kubernetes",
 		"Defines the cluster and service registry that this Istiod instance is belongs to").Get()
 
-	ExternalIstiod = env.RegisterBoolVar("EXTERNAL_ISTIOD", false,
+	ExternalIstiod = env.RegisterBoolVar("EXTERNAL_ISTIOD", true,
```

Comment:
I didn't keep up with the leader election changes, but IIUC, turning this on means we  will join LE to patch the `istio-ca-root-cert` configmap. We don't differentiate a "istiodless remote" from a "remote primary" cluster, but our leader election logic will make it so the "most local" istiod always wins. _Is this correct?_

We also have a "webhook cert patcher". and a "validating webhook controller". Those may need to be guarded by leader election as well?

Reply:
> We don't differentiate a "istiodless remote" from a "remote primary" cluster, but our leader election logic will make it so the "most local" istiod always wins. Is this correct?

Yes, that is correct.

> We also have a "webhook cert patcher". and a "validating webhook controller". Those may need to be guarded by leader election as well?

I had assumed that the istiod that gets the leader lock would be the one doing all these things, but maybe I'm missing something? Can you point me at the code where I should check and fix if necessary?

Reply:
https://github.com/istio/istio/blob/276fb292f30e3f20cc036fabba6328b76291c781/pilot/pkg/serviceregistry/kube/controller/multicluster.go#L235

There are a flew blocks in this method that check `features.ExternalIstiod` which. would be affected by this PR.